python/pyspark/ml/torch/distributor.py

```diff
@@ -554,7 +568,7 @@ def set_gpus(context: "BarrierTaskContext") -> None:
                     pass
 
             if context.partitionId() == 0:
-                yield output
+                yield pd.DataFrame(data={"output": [cloudpickle.dumps(output)]})
```

Comment:
Q: Did you test what happens if the output exceeding 2GB ? I guess it is not supported. Shall we split it in the case ?

Reply:
since the output dataframe only contains one row, I think `DataFrame.collect` can not properly handle it.

Reply:
> since the output dataframe only contains one row

We can make it contains multiple rows, this is doable.

Reply:
sounds good, break the row into multi rows in partition 0, and then concat them in driver